docs/metrics.md

```diff
@@ -0,0 +1,161 @@
+  | thanos_bucket_uiblocks_meta_modified                         | gauge     | Number of blocks whose metadata changed.                                                                                                                                            |
+  | thanos_bucket_uiblocks_meta_sync_duration_seconds            | histogram | Duration of the blocks metadata synchronization in seconds.                                                                                                                         |
+  | thanos_bucket_uiblocks_meta_sync_failures                    | counter   | Total blocks metadata synchronization failures.                                                                                                                                     |
+  | thanos_bucket_uiblocks_meta_synced                           | gauge     | Number of block metadata synced.                                                                                                                                                    |
+  | thanos_bucket_uiblocks_meta_syncs                            | counter   | Total blocks metadata synchronization attempts.                                                                                                                                     |
+
+### List of Metrics Exported By Object Storage
+- Thanos leverages object storage solutions such as Minio, Amazon S3, Google Cloud Storage, etc.
+
+  | Metric Name                                        | Type      | Description                                                                                                                                                          |
+  |----------------------------------------------------|-----------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+  | thanos_objstore_bucket_last_successful_upload_time | gauge     | Second timestamp of the last successful upload to the bucket.                                                                                                        |
+  | thanos_objstore_bucket_operation_duration_seconds  | histogram | Duration of successful operations against the bucket.                                                                                                                |
+  | thanos_objstore_bucket_operation_failures_total    | counter   | Total number of operations against a bucket that failed, but were not expected to fail in certain way from caller perspective. Those errors have to be investigated. |
+  | thanos_objstore_bucket_operations_total            | counter   | Total number of all attempted operations against a bucket.                                                                                                           |
+
+### List of Metrics Exported By Thanos Replicator
```

Comment:
What is the Thanos Replicator? IMHO, this component doesn't exist. It's a command from the Thanos Bucket Tool and should be there together with that command's docs.

Reply:
I disagree with this. The "general metrics" sections are for metrics exposed by more than one component of Thanos. Even though the Thanos bucket tool is not a "component" like others, it's its own separate thing and has its own documentation. So the metrics should be there, close to all the other information about it.

Reply:
Alright, Noted. 
I intend to write the Thanos bucket tool metrics to the `thanos/docs/components/tools,md` file.